llumnix/llumlet/local_migration_scheduler.py

```diff
@@ -39,35 +44,65 @@ def get_migrate_out_request(self, min_request_len=0, max_request_len=np.inf) ->
     # and only selects request that migrates from the prefill instance to the decoding instance.
     def get_ready_migration_request(self, min_request_len, max_request_len):
         running: List[LlumnixRequest] = self.backend_engine.get_running_queue()
+        target_request: LlumnixRequest = None
         for request in reversed(running):
+            if request.request_id in self.inflight_migrating_request:
+                continue
+
             if request.output_len >= request.expected_steps \
                 and request.inference_type == RequestInferenceType.DECODE \
                 and min_request_len <= request.request_len <= max_request_len:
-                return request
-        return None
+                target_request = request
+                break
+
+        if target_request:
+            self.inflight_migrating_request.add(target_request.request_id)
+
+        return target_request
 
     def get_last_running_request(self, min_request_len, max_request_len):
         running: List[LlumnixRequest] = self.backend_engine.get_running_queue()
+        target_request: LlumnixRequest = None
+
         for request in reversed(running):
+            if request.request_id in self.inflight_migrating_request:
+                continue
+
             if request.inference_type == RequestInferenceType.DECODE \
                 and min_request_len <= request.request_len <= max_request_len:
-                return request
-        return None
+                target_request=request
+                break
+
+        if target_request:
+            self.inflight_migrating_request.add(target_request.request_id)
```

Comment:
why still have inflight_migrating_request

Reply:
The inflight_migrating_request in **local_migartion_scheduler** is used to record which requests are currently being migrated within the current llumlet. It is used to avoid migrating a reuest twice.